R/bind_databases.R

```diff
@@ -0,0 +1,83 @@
+#' Bind multiple traits.build data objects into a single data object
+#'
+#' `bind_databases` binds all the listed studies into a single traits.build
+#' database object as a large list.
+#'
+#' @param ... Arguments passed to other functions
+#' @param databases List of traits.build databases to bind together (lists of tibbles)
+#'
+#' @return Compiled database as a single large list
+#' @importFrom rlang .data
+#' @export
+bind_databases <- function(database_1, ...) {
+  
+  # List of databases to combine
+  databases = list(database_1, ...)
+  
+  # Capture names of databases input; required below to assign names to databases in list
+  database_names <- match.call() %>% as.character()
+  database_names <- database_names[-1]
+  
+  combine <- function(name, databases) {
+    dplyr::bind_rows(lapply(databases, "[[", name)) %>% dplyr::distinct()
+  }
+  
+  # Bind sources and remove duplicates
+  sources <- databases %>% lapply("[[", "sources")
+  keys <- sources %>% lapply(names) %>% unlist() %>% unique() %>% sort()
+  sources <- sources %>% purrr::reduce(c)
+  sources <- sources[keys]
+  
+  definitions <- databases %>% lapply("[[", "definitions") %>% purrr::reduce(c)
+  definitions <- definitions[!duplicated(names(definitions))]
+  definitions <- definitions[sort(names(definitions))]
+  
+  # Drop null datasets
+  databases[sapply(databases, is.null)] <- NULL
+  
+  # Assign names to databases
+  names(databases) <- database_names
```

Comment:
@dfalster This step doesn't seem to be necessary because the names vanish as the datasets are bound together and I get the same output for austraits tests and building austraits.build without it. If it isn't needed, neither are lines 17/18 above